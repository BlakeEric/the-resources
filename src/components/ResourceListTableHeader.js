import React from "react"

export default function ResourceListTableHeader(props) {

  /*
   * Display up or down icon based on key and current order in state
   */
  const displaySortIcon = (keyName) => {
    if (props.orderBy === keyName) {
      if (props.order === "DESC") {
        return '↑'
      }
      return '↓'
    }
    return ''
  }

  /*
   * convert string to uppercase sentence format
   */
  const formatColumnTitle = (string) => {
    let words = string.replace( /([A-Z])/g, " $1" ).split(" ");
    for (let i = 0, x = words.length; i < x; i++) {
      words[i] = words[i][0].toUpperCase() + words[i].substr(1);
    }
    return words.join(" ");
  }

  return (
    <thead>
      <tr>
        {props.keys.map(keyName => {
          return (
            <th key={keyName} className={props.orderBy === keyName ? "active" : ""}>
              <button onClick={() => props.handleFilterClick(keyName)}>
                {formatColumnTitle(keyName)} {displaySortIcon(keyName)}
              </button>
            </th>
          )
        })}
      </tr>
    </thead>
  )
}
