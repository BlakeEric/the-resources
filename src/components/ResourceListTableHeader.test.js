import React from "react"
import renderer from "react-test-renderer"
import ResourceListTableHeader from "./ResourceListTableHeader"

let mockProps = {
  keys: ["name", "skillLevel", "format", "focus"],
  orderBy: "name",
  order: "ASC",
  handleFilterClick: () => {}
}

describe("ResourceListTableHeader", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<ResourceListTableHeader {...mockProps} />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it("Adds appropriate className and icon to current orderBy key", () => {

    mockProps.orderBy = "skillLevel"
    mockProps.order = "DESC"

    const tree = renderer.create(<ResourceListTableHeader {...mockProps} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
