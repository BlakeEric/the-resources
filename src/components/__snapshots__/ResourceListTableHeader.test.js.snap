// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResourceListTableHeader Adds appropriate className and icon to current orderBy key 1`] = `
<thead
  css={
    Object {
      "map": undefined,
      "name": "1soaygs",
      "next": undefined,
      "styles": "
  thead{
    padding: 20px;
  }
  th {
    font-family: Lato,sans-serif;
  }
  th button {
    background: white;
    padding: 0;
    box-shadow: none;
    border: none;
    cursor: pointer;
    border-radius: 5px;
    &:hover, &:focus {
      text-decoration: underline;
    }
  }
  th.active button {
    font-weight: 400;
    background-color: #0acc9e;
    background-image: linear-gradient(to bottom right, #37e6bd, #0acc9e);
    color: white;
    box-shadow: 0 2px 2px 0 rgba(0,0,0,0.05);
    padding: 0 0.4025rem;
    &:hover, &:focus {
      text-decoration: none;
    }
  }
  th {
    width: 25%;
    padding-top: 0.25rem;
    padding-bottom: calc(0.25rem - 1px);
  }
",
      "toString": [Function],
    }
  }
>
  <tr>
    <th
      className=""
    >
      <button
        onClick={[Function]}
      >
        Name
         
        
      </button>
    </th>
    <th
      className="active"
    >
      <button
        onClick={[Function]}
      >
        Skill Level
         
        ↑
      </button>
    </th>
    <th
      className=""
    >
      <button
        onClick={[Function]}
      >
        Format
         
        
      </button>
    </th>
    <th
      className=""
    >
      <button
        onClick={[Function]}
      >
        Focus
         
        
      </button>
    </th>
  </tr>
</thead>
`;

exports[`ResourceListTableHeader renders correctly 1`] = `
<thead
  css={
    Object {
      "map": undefined,
      "name": "1soaygs",
      "next": undefined,
      "styles": "
  thead{
    padding: 20px;
  }
  th {
    font-family: Lato,sans-serif;
  }
  th button {
    background: white;
    padding: 0;
    box-shadow: none;
    border: none;
    cursor: pointer;
    border-radius: 5px;
    &:hover, &:focus {
      text-decoration: underline;
    }
  }
  th.active button {
    font-weight: 400;
    background-color: #0acc9e;
    background-image: linear-gradient(to bottom right, #37e6bd, #0acc9e);
    color: white;
    box-shadow: 0 2px 2px 0 rgba(0,0,0,0.05);
    padding: 0 0.4025rem;
    &:hover, &:focus {
      text-decoration: none;
    }
  }
  th {
    width: 25%;
    padding-top: 0.25rem;
    padding-bottom: calc(0.25rem - 1px);
  }
",
      "toString": [Function],
    }
  }
>
  <tr>
    <th
      className="active"
    >
      <button
        onClick={[Function]}
      >
        Name
         
        ↓
      </button>
    </th>
    <th
      className=""
    >
      <button
        onClick={[Function]}
      >
        Skill Level
         
        
      </button>
    </th>
    <th
      className=""
    >
      <button
        onClick={[Function]}
      >
        Format
         
        
      </button>
    </th>
    <th
      className=""
    >
      <button
        onClick={[Function]}
      >
        Focus
         
        
      </button>
    </th>
  </tr>
</thead>
`;
